Competition2017/src/competition/subsystems/shooter_belt/commands/RunBeltIfWheelAtSpeedCommand.java

```diff
@@ -0,0 +1,37 @@
+package competition.subsystems.shooter_belt.commands;
+
+import org.apache.log4j.Logger;
+
+import competition.subsystems.shooter_belt.ShooterBeltSubsystem;
+import competition.subsystems.shooter_wheel.ShooterWheelSubsystem;
+import xbot.common.command.BaseCommand;
+
+public class RunBeltIfWheelAtSpeedCommand extends BaseCommand {
+
+    private static Logger log = Logger.getLogger(RunBeltIfWheelAtSpeedCommand.class);
+    
+    final ShooterBeltSubsystem beltSubsystem;
+    public ShooterWheelSubsystem shooterWheelSubsystem;
+    final double targetBeltSpeed;
+    
+    public RunBeltIfWheelAtSpeedCommand(ShooterBeltSubsystem beltSubsystem){
+        this.beltSubsystem = beltSubsystem;
+        this.requires(beltSubsystem);
+        targetBeltSpeed = beltSubsystem.getBeltTargetSpeed();
```

Comment:
This feels wrong. The command saves the current speed, then stops it and occasionally uses that speed to start it again? So you're using the subsystem's target speed as a way of passing an argument?